ui/components/multichain/account-list-item/index.js

```diff
@@ -0,0 +1,218 @@
+import { SECONDARY, PRIMARY } from '../../../helpers/constants/common';
+import { findKeyringForAddress } from '../../../ducks/metamask/metamask';
+import Tooltip from '../../ui/tooltip/tooltip';
+
+const MAXIMUM_CURRENCY_DECIMALS = 3;
+
+function getLabel(keyring = {}, t) {
+  const { type } = keyring;
+  switch (type) {
+    case HardwareKeyringTypes.qr:
+      return HardwareKeyringNames.qr;
+    case HardwareKeyringTypes.imported:
+      return t('imported');
+    case HardwareKeyringTypes.trezor:
+      return HardwareKeyringNames.trezor;
+    case HardwareKeyringTypes.ledger:
+      return HardwareKeyringNames.ledger;
+    case HardwareKeyringTypes.lattice:
+      return HardwareKeyringNames.lattice;
+    default:
+      return null;
+  }
+}
+
+export const AccountListItem = ({
+  identity,
+  selected = false,
+  onClick,
+  closeMenu,
+  connectedAvatar,
+  connectedAvatarName,
+}) => {
+  const t = useI18nContext();
+  const [accountOptionsMenuOpen, setAccountOptionsMenuOpen] = useState(false);
+  const ref = useRef(false);
+  const keyring = useSelector((state) =>
+    findKeyringForAddress(state, identity.address),
+  );
+  const label = getLabel(keyring, t);
+
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const { blockExplorerUrl } = rpcPrefs;
+  const blockExplorerUrlSubTitle = getURLHostName(blockExplorerUrl);
+
+  return (
+    <Box
+      display={DISPLAY.FLEX}
+      padding={4}
+      gap={2}
+      backgroundColor={selected ? Color.primaryMuted : Color.transparent}
+      className={classnames('multichain-account-list-item', {
+        'multichain-account-list-item--selected': selected,
+      })}
+      as="button"
+      onClick={(e) => {
+        e.preventDefault();
+        // Without this check, the account will be selected after
+        // the account options menu closes
+        if (!accountOptionsMenuOpen) {
+          onClick();
+        }
+      }}
+    >
+      {selected && (
+        <Box
+          className="multichain-account-list-item__selected-indicator"
+          borderRadius={BorderRadius.pill}
+          backgroundColor={Color.primaryDefault}
+        />
+      )}
+      <AvatarAccount size={Size.SM} address={identity.address}></AvatarAccount>
+      <Box
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+        className="multichain-account-list-item__content"
+      >
+        <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.COLUMN}>
+          <Box
+            display={DISPLAY.FLEX}
+            justifyContent={JustifyContent.spaceBetween}
+            gap={2}
+          >
+            <Text ellipsis>
+              <Tooltip
+                title={identity.name}
+                position="bottom"
+                wrapperClassName="multichain-account-list-item__tooltip"
+              >
+                {identity.name}
+              </Tooltip>
+            </Text>
```

Comment:
This is breaking the story, since the HTML structure looks like <p><div><div></p>.
```suggestion
            <Text ellipsis as="div">
              <Tooltip
                title={identity.name}
                position="bottom"
                wrapperClassName="multichain-account-list-item__tooltip"
              >
                {identity.name}
              </Tooltip>
            </Text>
```

Reply:
Good shout!